apps/server/src/modules/files-storage/service/files-storage-upload.service.spec.ts

```diff
@@ -127,40 +125,119 @@ describe('FilesStorageService upload methods', () => {
 
-		it('should call getFileRecordsOfParent with correct params', async () => {
-			const { params, file, userId, getFileRecordsOfParentSpy } = createUploadFileParams();
+		describe('WHEN file records of parent, file record repo save and get mime type are successfull', () => {
+			const setup = () => {
+				const { params, file, fileSize, userId, fileRecord, expectedFileRecord, detectedMimeType, readable } =
+					createUploadFileParams();
+
+				const getFileRecordsOfParentSpy = jest
+					.spyOn(service, 'getFileRecordsOfParent')
+					.mockResolvedValue([[fileRecord], 1]);
+
+				const getMimeTypeSpy = jest
+					.spyOn(StreamMimeType, 'getMimeType')
+					.mockResolvedValueOnce({ mime: detectedMimeType, stream: readable as unknown as undefined });
+
+				// The fileRecord._id must be set by fileRecordRepo.save. Otherwise createPath fails.
+				// eslint-disable-next-line @typescript-eslint/require-await
+				fileRecordRepo.save.mockImplementation(async (fr) => {
+					if (fr instanceof FileRecord && !fr._id) {
+						fr._id = new ObjectId();
+					}
+				});
+
+				return { params, file, userId, getFileRecordsOfParentSpy, getMimeTypeSpy, fileSize, expectedFileRecord };
+			};
 
-			await service.uploadFile(userId, params, file);
+			it('should call getMimeType with correct params', async () => {
+				const { params, file, userId, getMimeTypeSpy } = setup();
 
-			expect(getFileRecordsOfParentSpy).toHaveBeenCalledWith(params.parentId);
-		});
+				await service.uploadFile(userId, params, file);
 
-		it('should call fileRecordRepo.save twice with correct params', async () => {
-			const { params, file, fileSize, userId, expectedFileRecord } = createUploadFileParams();
+				expect(getMimeTypeSpy).toHaveBeenCalledWith(file.data, { strict: true });
+			});
 
-			await service.uploadFile(userId, params, file);
+			it('should call getFileRecordsOfParent with correct params', async () => {
+				const { params, file, userId, getFileRecordsOfParentSpy } = setup();
 
-			expect(fileRecordRepo.save).toHaveBeenCalledTimes(2);
+				await service.uploadFile(userId, params, file);
+
+				expect(getFileRecordsOfParentSpy).toHaveBeenCalledWith(params.parentId);
+			});
 
-			expect(fileRecordRepo.save).toHaveBeenLastCalledWith(
-				expect.objectContaining({
-					...expectedFileRecord,
-					size: fileSize,
-					createdAt: expect.any(Date),
-					updatedAt: expect.any(Date),
-				})
-			);
+			it('should call fileRecordRepo.save twice with correct params', async () => {
+				const { params, file, fileSize, userId, expectedFileRecord } = setup();
+
+				await service.uploadFile(userId, params, file);
+
+				expect(fileRecordRepo.save).toHaveBeenCalledTimes(2);
+
+				expect(fileRecordRepo.save).toHaveBeenLastCalledWith(
+					expect.objectContaining({
+						...expectedFileRecord,
+						size: fileSize,
+						createdAt: expect.any(Date),
+						updatedAt: expect.any(Date),
+					})
+				);
+			});
+
+			it('should correctly set file size', async () => {
```

Comment:
This test is already contained in the above test. I'd say we don't need a separate test for that.

Reply:
Mocks should be located in the setup functions of the scenario